services/skus/datastore.go

```diff
@@ -542,6 +543,27 @@ func (pg *Postgres) CheckExpiredCheckoutSession(orderID uuid.UUID) (bool, string
 	return true, *checkoutSession, nil
 }
 
+func (pg *Postgres) ExternalIDExists(ctx context.Context, externalID string) (bool, error) {
+	var (
+		ok  bool
+		md  datastore.Metadata
+		err error
+	)
+
+	err = pg.RawDB().Get(&md, `
+		SELECT metadata
+		FROM orders
+		WHERE metadata->>'externalID' = $1 AND metadata is not null
+	`, externalID)
```

Comment:
i'm not familiar with this function but does it prevent SQL injection attacks? since it seems externalID could be anything

Reply:
so `RawDB()` is returning a `*sqlx.DB`. `DB.Get` uses `DB.QueryRowx` which in turn calls `Query` on the stdlib `*sql.DB`. https://go.dev/doc/database/sql-injection explains that under the hood this turns the SQL statement into a prepared statement and sends it along with the parameter to the database avoiding SQL injection attacks